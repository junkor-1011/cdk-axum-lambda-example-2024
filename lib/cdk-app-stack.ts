import { Stack, type StackProps } from 'aws-cdk-lib';
import type { Construct } from 'constructs';

import { ApiSystemWithLayer } from './constructs/api-sytem-with-layer';
import { ApiSystemWithoutLayer } from './constructs/api-sytem-without-layer';

export class CdkAppStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    new ApiSystemWithLayer(this, 'api-system-with-layer', {
      name: 'api-system-with-layer',
    });

    new ApiSystemWithoutLayer(this, 'api-system-without-layer', {
      name: 'api-system-without-layer',
    });
  }
}
