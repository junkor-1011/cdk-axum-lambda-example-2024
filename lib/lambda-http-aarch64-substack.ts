import { NestedStack, type NestedStackProps } from 'aws-cdk-lib';
import { Architecture } from 'aws-cdk-lib/aws-lambda';
import type { Construct } from 'constructs';

import { ApiSystemUsingLambdaHttp } from './constructs/api-system-using-lambda-http';

export class LambdaHttpARM64ExampleStack extends NestedStack {
  constructor(scope: Construct, id: string, props?: NestedStackProps) {
    super(scope, id, props);

    new ApiSystemUsingLambdaHttp(this, 'api-system-using-lambda-http-aarch64', {
      name: 'api-system-using-lambda-http-aarch64',
      architecture: Architecture.ARM_64,
    });
  }
}
