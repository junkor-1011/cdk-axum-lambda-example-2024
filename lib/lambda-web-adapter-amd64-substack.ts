import { NestedStack, type NestedStackProps } from 'aws-cdk-lib';
import { Architecture } from 'aws-cdk-lib/aws-lambda';
import type { Construct } from 'constructs';

import { ApiSystemUsingLambdaWebAdapter } from './constructs/api-system-using-lambda-webadapter-layer';

export class LambdaWebAdapterAMD64ExampleStack extends NestedStack {
  constructor(scope: Construct, id: string, props?: NestedStackProps) {
    super(scope, id, props);

    new ApiSystemUsingLambdaWebAdapter(
      this,
      'api-system-using-webadapter-layer-amd64',
      {
        name: 'api-system-using-webadapter-layer-amd64',
        architecture: Architecture.X86_64,
      },
    );
  }
}
