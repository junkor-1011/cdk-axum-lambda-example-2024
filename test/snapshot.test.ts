import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { expect, test } from 'vitest';
import { CdkAppStack } from '../lib/cdk-app-stack';
import { ignoreAssetHashSerializer } from './plugins/ignore-asset-hash';

test('snapshot test(root stack)', () => {
  const app = new cdk.App();
  const stack = new CdkAppStack(app, 'TestRootStack');
  const template = Template.fromStack(stack).toJSON();

  expect.addSnapshotSerializer(ignoreAssetHashSerializer);
  expect(template).toMatchSnapshot();

  for (const child of stack.node.children.filter(
    (child) => child instanceof cdk.NestedStack,
  )) {
    expect(Template.fromStack(child as cdk.NestedStack)).toMatchSnapshot();
  }
});
