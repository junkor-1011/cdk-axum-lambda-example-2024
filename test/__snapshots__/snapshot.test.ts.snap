// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot test 1`] = `
{
  "Outputs": {
    "apisystemwithlayerAPIusingaxumEndpointAFBBC82B": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "apisystemwithlayerAPIusingaxumAE5651FB",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "apisystemwithlayerAPIusingaxumDeploymentStageprod525A5E6F",
            },
            "/",
          ],
        ],
      },
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpEndpoint6152FD7C": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttpDeploymentStageprod2968ED52",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "apisystemwithlayerAPIusingaxumAE5651FB": {
      "Properties": {
        "Name": "API using axum",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "apisystemwithlayerAPIusingaxumANY741C500F": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "apisystemwithlayeraxumlambdaserver2667726B",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Fn::GetAtt": [
            "apisystemwithlayerAPIusingaxumAE5651FB",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "apisystemwithlayerAPIusingaxumAE5651FB",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "apisystemwithlayerAPIusingaxumANYApiPermissionTestStackapisystemwithlayerAPIusingaxum3BE90064ANYF1C0259D": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "apisystemwithlayeraxumlambdaserver2667726B",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "apisystemwithlayerAPIusingaxumAE5651FB",
              },
              "/",
              {
                "Ref": "apisystemwithlayerAPIusingaxumDeploymentStageprod525A5E6F",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "apisystemwithlayerAPIusingaxumANYApiPermissionTestTestStackapisystemwithlayerAPIusingaxum3BE90064ANY06BACD4B": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "apisystemwithlayeraxumlambdaserver2667726B",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "apisystemwithlayerAPIusingaxumAE5651FB",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "apisystemwithlayerAPIusingaxumAccount8CC61821": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "apisystemwithlayerAPIusingaxumAE5651FB",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "apisystemwithlayerAPIusingaxumCloudWatchRole021999ED",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "apisystemwithlayerAPIusingaxumCloudWatchRole021999ED": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "apisystemwithlayerAPIusingaxumDeployment53C53BE84bfc9e4692ece98c5ed2a4243aff9e9d": {
      "DependsOn": [
        "apisystemwithlayerAPIusingaxumproxyANYE9763BE6",
        "apisystemwithlayerAPIusingaxumproxy15678E58",
        "apisystemwithlayerAPIusingaxumANY741C500F",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "apisystemwithlayerAPIusingaxumAE5651FB",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "apisystemwithlayerAPIusingaxumDeploymentStageprod525A5E6F": {
      "DependsOn": [
        "apisystemwithlayerAPIusingaxumAccount8CC61821",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "apisystemwithlayerAPIusingaxumDeployment53C53BE84bfc9e4692ece98c5ed2a4243aff9e9d",
        },
        "RestApiId": {
          "Ref": "apisystemwithlayerAPIusingaxumAE5651FB",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "apisystemwithlayerAPIusingaxumproxy15678E58": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "apisystemwithlayerAPIusingaxumAE5651FB",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": {
          "Ref": "apisystemwithlayerAPIusingaxumAE5651FB",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "apisystemwithlayerAPIusingaxumproxyANYApiPermissionTestStackapisystemwithlayerAPIusingaxum3BE90064ANYproxy32BBB68D": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "apisystemwithlayeraxumlambdaserver2667726B",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "apisystemwithlayerAPIusingaxumAE5651FB",
              },
              "/",
              {
                "Ref": "apisystemwithlayerAPIusingaxumDeploymentStageprod525A5E6F",
              },
              "/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "apisystemwithlayerAPIusingaxumproxyANYApiPermissionTestTestStackapisystemwithlayerAPIusingaxum3BE90064ANYproxy76B3FDA5": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "apisystemwithlayeraxumlambdaserver2667726B",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "apisystemwithlayerAPIusingaxumAE5651FB",
              },
              "/test-invoke-stage/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "apisystemwithlayerAPIusingaxumproxyANYE9763BE6": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "apisystemwithlayeraxumlambdaserver2667726B",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "apisystemwithlayerAPIusingaxumproxy15678E58",
        },
        "RestApiId": {
          "Ref": "apisystemwithlayerAPIusingaxumAE5651FB",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "apisystemwithlayerLambdaWebAdapter977375BB": {
      "Properties": {
        "CompatibleArchitectures": [
          "x86_64",
        ],
        "Content": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "HASH-REPLACED.zip",
        },
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "apisystemwithlayeraxumlambdaserver2667726B": {
      "DependsOn": [
        "apisystemwithlayeraxumlambdaserverServiceRoleDefaultPolicy9E51AE75",
        "apisystemwithlayeraxumlambdaserverServiceRole30D819AE",
      ],
      "Properties": {
        "Architectures": [
          "x86_64",
        ],
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "HASH-REPLACED.zip",
        },
        "Environment": {
          "Variables": {
            "PORT": "3000",
            "RUST_LOG": "debug",
          },
        },
        "FunctionName": "axum-server-function",
        "Handler": "bootstrap",
        "Layers": [
          {
            "Ref": "apisystemwithlayerLambdaWebAdapter977375BB",
          },
        ],
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "apisystemwithlayeraxumlambdaserverServiceRole30D819AE",
            "Arn",
          ],
        },
        "Runtime": "provided.al2023",
        "Timeout": 10,
        "TracingConfig": {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "apisystemwithlayeraxumlambdaserverServiceRole30D819AE": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "apisystemwithlayeraxumlambdaserverServiceRoleDefaultPolicy9E51AE75": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "apisystemwithlayeraxumlambdaserverServiceRoleDefaultPolicy9E51AE75",
        "Roles": [
          {
            "Ref": "apisystemwithlayeraxumlambdaserverServiceRole30D819AE",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C": {
      "Properties": {
        "Name": "API using axum and lambda-http",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpANY1C9895C4": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "apisystemwithoutlayeraxumlambdahttp047E1DA1",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Fn::GetAtt": [
            "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpANYApiPermissionTestStackapisystemwithoutlayerAPIusingaxumandlambdahttp41E53225ANY2DD51AF5": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "apisystemwithoutlayeraxumlambdahttp047E1DA1",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
              },
              "/",
              {
                "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttpDeploymentStageprod2968ED52",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpANYApiPermissionTestTestStackapisystemwithoutlayerAPIusingaxumandlambdahttp41E53225ANY96DDC827": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "apisystemwithoutlayeraxumlambdahttp047E1DA1",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpAccountB52F88C4": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "apisystemwithoutlayerAPIusingaxumandlambdahttpCloudWatchRole498B92AF",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpCloudWatchRole498B92AF": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpDeploymentF7350149c7df13b76424d3a2c674e1e555953653": {
      "DependsOn": [
        "apisystemwithoutlayerAPIusingaxumandlambdahttpproxyANYFB2AA70A",
        "apisystemwithoutlayerAPIusingaxumandlambdahttpproxyB9CEA8DC",
        "apisystemwithoutlayerAPIusingaxumandlambdahttpANY1C9895C4",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpDeploymentStageprod2968ED52": {
      "DependsOn": [
        "apisystemwithoutlayerAPIusingaxumandlambdahttpAccountB52F88C4",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttpDeploymentF7350149c7df13b76424d3a2c674e1e555953653",
        },
        "RestApiId": {
          "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpproxyANYApiPermissionTestStackapisystemwithoutlayerAPIusingaxumandlambdahttp41E53225ANYproxy305CB808": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "apisystemwithoutlayeraxumlambdahttp047E1DA1",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
              },
              "/",
              {
                "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttpDeploymentStageprod2968ED52",
              },
              "/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpproxyANYApiPermissionTestTestStackapisystemwithoutlayerAPIusingaxumandlambdahttp41E53225ANYproxyE335D4AC": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "apisystemwithoutlayeraxumlambdahttp047E1DA1",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
              },
              "/test-invoke-stage/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpproxyANYFB2AA70A": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "apisystemwithoutlayeraxumlambdahttp047E1DA1",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttpproxyB9CEA8DC",
        },
        "RestApiId": {
          "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "apisystemwithoutlayerAPIusingaxumandlambdahttpproxyB9CEA8DC": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": {
          "Ref": "apisystemwithoutlayerAPIusingaxumandlambdahttp4AFB0D2C",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "apisystemwithoutlayeraxumlambdahttp047E1DA1": {
      "DependsOn": [
        "apisystemwithoutlayeraxumlambdahttpServiceRoleDefaultPolicy5873A8F0",
        "apisystemwithoutlayeraxumlambdahttpServiceRoleD1DB66E3",
      ],
      "Properties": {
        "Architectures": [
          "x86_64",
        ],
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "HASH-REPLACED.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_LAMBDA_HTTP_IGNORE_STAGE_IN_PATH": "true",
            "RUST_LOG": "debug",
          },
        },
        "FunctionName": "axum-lambda-http",
        "Handler": "bootstrap",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "apisystemwithoutlayeraxumlambdahttpServiceRoleD1DB66E3",
            "Arn",
          ],
        },
        "Runtime": "provided.al2023",
        "Timeout": 10,
        "TracingConfig": {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "apisystemwithoutlayeraxumlambdahttpServiceRoleD1DB66E3": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "apisystemwithoutlayeraxumlambdahttpServiceRoleDefaultPolicy5873A8F0": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "apisystemwithoutlayeraxumlambdahttpServiceRoleDefaultPolicy5873A8F0",
        "Roles": [
          {
            "Ref": "apisystemwithoutlayeraxumlambdahttpServiceRoleD1DB66E3",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
